zstd_ctx.go

```diff
@@ -1,6 +1,7 @@
 package zstd
 
 /*
+#cgo CFLAGS: -O3
```

Comment:
👍 If we aren't already compiling (our external?) zstd library with -O3, this is probably a good idea. But I don't know if that's the case.